Add a short introduction to ZHA

Hedda opened the pull request:
## Proposed change

Adds a short ZHA integration introduction with Zigbee diction and essential limitations as summary meant for new ZHA end users.

I think this will help explain the most essential questions that many potential ZHA users (and also users new to Zigbee) commonly have and ask before they buy an adapter and install the ZHA integration or are first asked directly after they have installed ZHA.

The purpose of this short introduction is to help new users quickly get up to speed on the hardware requirements for ZHA as well as mention the minimum essential Zigbee terminology that is really needed to get easily up and running with the ZHA integration.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

Hedda committed:
Add a short introduction to ZHA

Add a short ZHA integration introduction with Zigbee diction and essential limitations as a summary meant for new end users.

frenck commented:
You already had a PR open for this. I suggest you close this one and update the existing one instead. Thanks 👍

Hedda commented:
Prefer to keep this if OK. Assume the other one was ignored for so long because got rewritten too much so see this as fresh start.

frenck reviewed (changes requested):
> Prefer to keep this if OK.

That is not OK. Spawning more PRs doesn't help, it also make history lost.
Please just update your previous PR.s

Hedda commented:
What does it matter? No one seems to review ZHA docs PRs anyway and that is mostly what that history shows.

frenck commented:
It matters because there is a review history there, providing context. There is absolutely no single reason to create new PRs.

It also was done with all your PRs without any reason provided at all, nor was there any motivation or indication that there was not other option besides creating a new PR.

Please close these PRs and update the previous PRs instead. 

Thanks 👍 

../Frenck

krati-ag207 commented:
Please assign me this issue

frenck commented:
@krati-ag207 this is not an issue... it is a pull request...

frenck commented:
@Hedda could you please take a look an resolve to open review comment, thanks!

Hedda commented:
@frenck The old PR was locked by github-actions bot so I can not reopoen it myself, please see -> https://github.com/home-assistant/home-assistant.io/pull/22260

Hedda committed:
Merge branch 'current' into patch-79

Hedda commented:
https://github.com/home-assistant/home-assistant.io/pull/22260 is still locked so should we move forward with https://github.com/home-assistant/home-assistant.io/pull/24494 or?

Hedda committed:
Split limitations section in ZHA introduction

Split the limitations into a separate section under the ZHA introduction.

Hedda committed:
Merge branch 'current' into patch-79

Hedda committed:
Update zha.markdown

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -67,6 +67,22 @@ There is currently support for the following device types within Home Assistant:
 
+## Introduction
+
+ZHA integration follows the official Zigbee specification standards and uses one Zigbee Coordinator to set up and control a Zigbee network. Once ZHA has been installed and the Zigbee Coordinator is configured you will be able to directly join/pair Zigbee Router Device and Zigbee End Device products to that Zigbee network, regardless of the manufacturer and brand of that Zigbee-based product. Plea ...
```
The first response to the first sentence: What is a coordinator? (Also, why is it capitalized?)

Hedda replied:
OK, I have rewritten much of the Introduction section for ZHA based on feedback.

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -67,6 +67,22 @@ There is currently support for the following device types within Home Assistant:
+
+ZHA integration follows the official Zigbee specification standards and uses one Zigbee Coordinator to set up and control a Zigbee network. Once ZHA has been installed and the Zigbee Coordinator is configured you will be able to directly join/pair Zigbee Router Device and Zigbee End Device products to that Zigbee network, regardless of the manufacturer and brand of that Zigbee-based product. Plea ...
+
+Before installing the integration you will need to buy and connect a Zigbee Coordinator adapter to the computer running the Home Assistant installation. The general recommendation is to buy a newer model of Zigbee Coordinator adapter hardware and to flash the latest firmware to it, the reason that is it will usually offer better interoperability with all functions of most Zigbee 3.0 compliant dev ...
```
I think the new user will absolutely get lost in this paragraph. While still not clean what a coordinator is, it now becomes an adapter too? And what is firmware? And how do I flash it onto my unknown coordinator?

In the last line, is now talks about Zigbee 3.0?

Hedda replied:
OK, I have rewritten much of the Introduction section for ZHA based on feedback.

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -67,6 +67,22 @@ There is currently support for the following device types within Home Assistant:
+
+Supported devices are Zigbee Router Devices, which are normally mains-powered that act as Zigbee signal repeaters within the Zigbee network mesh to extend its range and improve coverage, and Zigbee End Devices, which will not act as Zigbee signal repeaters and are usually but not always battery-operated sensors.
+
+### Limitations
```
I don't think limitation belong at the start of documentation. Know limitations are usually listed in their own chapter somewhere at the end.

For example, the fact you can only use a single coordinator, isn't relevant at this point of the documentation at all. Nor does it matter if your devices are already paired or not at this point (that part would better fit in a pairing paragraph).

Hedda replied:
OK, I moved the Limitation into its own section as suggested, so it is now under Troubleshooting as I think the users should be aware of the limitations before they start troubleshooting.

Hedda committed:
Merge branch 'current' into patch-79

Hedda committed:
Rewrote ZHA Introduction and moved Limitation

Rewrote the Introduction section for ZHA based on feedback and moved the Limitation section to under Troubleshooting as suggested.

Hedda commented:
OK, I rewritten the Introduction section for ZHA based on feedback and moved the Limitation section.

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -67,6 +67,18 @@ There is currently support for the following device types within Home Assistant:
+
+ZHA integration is a Zigbee gateway implementation that follows the standard Zigbee 3.0 specification as well as earlier revisions of the Zigbee Pro specification. In Zigbee, there are three different device types: Zigbee Coordinator (ZC), Zigbee Router (ZR), and Zigbee End Device (ZED). A Zigbee network always has one (and no more) Zigbee Coordinator, however, a Zigbee network can have multiple  ...
+
+A Zigbee Coordinator is the central device in a Zigbee network that manages and controls the network, and assigns addresses to other devices. It acts as a physical interface for the Zigbee wireless protocol, as well as being responsible for maintaining the Zigbee network topology and ensuring secure and efficient communication between Zigbee devices. Zigbee Router devices are mains-powered and wi ...
```
```suggestion
A Zigbee Coordinator is the central device in a Zigbee network that manages and controls the network. It acts as a physical interface for the Zigbee wireless protocol, as well as being responsible for maintaining the Zigbee network topology and ensuring secure and efficient communication between Zigbee devices. Zigbee Router devices are mains-powered and will act as Zigbee signal repeaters within the Zigbee network mesh to extend its range and improve coverage, while Zigbee End Device devices are usually battery-operated sensors that will not act as Zigbee signal repeaters.
```

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -67,6 +67,18 @@ There is currently support for the following device types within Home Assistant:
+
+A Zigbee Coordinator is the central device in a Zigbee network that manages and controls the network, and assigns addresses to other devices. It acts as a physical interface for the Zigbee wireless protocol, as well as being responsible for maintaining the Zigbee network topology and ensuring secure and efficient communication between Zigbee devices. Zigbee Router devices are mains-powered and wi ...
+
+In the case of ZHA, the Zigbee Coordinator is a radio adapter or module hardware, that hardware contains a microcontroller which comes with firmware that runs the Zigbee protocol stack that the ZHA integration uses to manage a Zigbee network and communicate wirelessly with Zigbee devices via an integrated low-power digital radio and antenna.
```
```suggestion
In the case of ZHA, the Zigbee Coordinator is a radio adapter or hardware module, that contains a microcontroller that runs the Zigbee protocol stack which the ZHA integration uses to manage and communicate with a Zigbee network and its devices.
```

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -67,6 +67,18 @@ There is currently support for the following device types within Home Assistant:
 
+## Introduction
+
+ZHA integration is a Zigbee gateway implementation that follows the standard Zigbee 3.0 specification as well as earlier revisions of the Zigbee Pro specification. In Zigbee, there are three different device types: Zigbee Coordinator (ZC), Zigbee Router (ZR), and Zigbee End Device (ZED). A Zigbee network always has one (and no more) Zigbee Coordinator, however, a Zigbee network can have multiple  ...
```
```suggestion
ZHA integration is a Zigbee gateway implementation that follows the standard Zigbee 3.0 specification (and earlier revisions). In Zigbee, there are three different device types: Zigbee Coordinator (ZC), Zigbee Router (ZR), and Zigbee End Device (ZED). A Zigbee network always has one (and no more) Zigbee Coordinator, however, a Zigbee network can have multiple Zigbee Routers and multiple Zigbee End Devices.
```

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -67,6 +67,18 @@ There is currently support for the following device types within Home Assistant:
+
+In the case of ZHA, the Zigbee Coordinator is a radio adapter or module hardware, that hardware contains a microcontroller which comes with firmware that runs the Zigbee protocol stack that the ZHA integration uses to manage a Zigbee network and communicate wirelessly with Zigbee devices via an integrated low-power digital radio and antenna.
+
+Therefore before installing the ZHA integration, you will need to buy and connect a Zigbee Coordinator radio adapter or module hardware, those usually come in the form of a USB dongle that plugs directly into the computer that is running the Home Assistant installation. The ZHA integration can work with many different "Zigbee Coordinator" adapters, however, be sure to read the respective sections ...
```
```suggestion
Before installing the ZHA integration, you will need to buy and connect a Zigbee Coordinator radio adapter or hardware module, those usually come in the form of a USB dongle that plugs directly into the computer that is running the Home Assistant installation. The ZHA integration can work with many different "Zigbee Coordinator" adapters, however, be sure to read the respective sections below about compatible Zigbee radio adapters and hardware modules.
```

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -67,6 +67,18 @@ There is currently support for the following device types within Home Assistant:
+
+Therefore before installing the ZHA integration, you will need to buy and connect a Zigbee Coordinator radio adapter or module hardware, those usually come in the form of a USB dongle that plugs directly into the computer that is running the Home Assistant installation. The ZHA integration can work with many different "Zigbee Coordinator" adapters, however, be sure to read the respective sections ...
+
+Once ZHA has been installed and the Zigbee Coordinator radio adapter or module is configured you will be able to directly join/pair Zigbee Router Device and Zigbee End Device products to that Zigbee network, regardless of the manufacturer and brand of that Zigbee-based product. Note that while it is generally recommended to buy Zigbee 3.0 compliant devices as those should, in theory, offer greate ...
```
```suggestion
Once ZHA has been set up and the Zigbee Coordinator radio adapter or module is configured, you will be able to directly join/pair any Zigbee device to the Zigbee network; regardless of the manufacturer and brand of that Zigbee-based product. Note, that while it is generally recommended to buy Zigbee 3.0 compliant devices as those should, in theory, offer greater interoperability, be sure of the sections about which devices are supported and exception handling.
```

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -423,7 +435,15 @@ Within ZHA is possible to use this backup and restore feature to migrate between
+
+Note that ZHA only supports connecting a single dedicated Zigbee Coordinator radio adapter or module with a single Zigbee network and that the Zigbee Coordinator cannot already be connected or used by any other application. Any devices that are or have previously been connected to another Zigbee implementation will also need to first be reset to their factory default settings before they can be p ...
+
+ZHA does currently not support devices that can only use the ZGP ("Zigbee Green Power") profile which is used in a few batteryless self-powered or energy harvesting devices, (such as for example; Philips Hue Click, Philips Hue Tap, and some "Friends of Hue" partnership switches). The reason for this is that ZGP profile support has not yet been implemented in the [zigpy](https://github.com/zigpy/z ...
```
```suggestion
ZHA does currently not support devices that can only use the ZGP ("Zigbee Green Power") profile which is used in a few batteryless self-powered or energy harvesting devices, (such as for example; Philips Hue Click, Philips Hue Tap, and some "Friends of Hue" partnership switches).
```

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -423,7 +435,15 @@ Within ZHA is possible to use this backup and restore feature to migrate between
+
+ZHA does currently not support devices that can only use the ZGP ("Zigbee Green Power") profile which is used in a few batteryless self-powered or energy harvesting devices, (such as for example; Philips Hue Click, Philips Hue Tap, and some "Friends of Hue" partnership switches). The reason for this is that ZGP profile support has not yet been implemented in the [zigpy](https://github.com/zigpy/z ...
+
+ZHA does not currently support devices that can only use the ZSE ("Zigbee Smart Energy") profile, that is however due to the "Zigbee SE" specification not being part of the standard Zigbee 3.0 specification and thus not implemented in most of the Zigbee protocol stacks that are commonly available Zigbee Coordinator radio adapters and modules.
```
```suggestion
ZHA does not currently support devices that can only use the ZSE ("Zigbee Smart Energy") profile, that is however due to the "Zigbee SE" specification not being part of the standard Zigbee 3.0 specification and thus not implemented in most of the Zigbee protocol stacks that are commonly available Zigbee Coordinator radio adapters and modules.
```

Hedda replied:
Was there a suggested change there? I am not ignoring but must be an error as do not see a commit suggestion. Please re-review and resubmit the suggested change there. Resolving this as can not see the requested change.

![image](https://user-images.githubusercontent.com/6320001/218690708-e7fb7e45-1158-4d63-a049-4a3d1a6e327a.png)

frenck replied:
I have no idea what happened or why I did that. Resolving it for that reason.

frenck reviewed (changes requested):
Getting there! Did a little bit of scratching to compact it a little more.

../Frenck

Hedda committed:
Update source/_integrations/zha.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Hedda committed:
Update source/_integrations/zha.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Hedda committed:
Update source/_integrations/zha.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Hedda committed:
Update source/_integrations/zha.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Hedda committed:
Update source/_integrations/zha.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Hedda committed:
Update source/_integrations/zha.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Hedda committed:
Merge branch 'current' into patch-79

frenck reviewed (approved):
Thanks, @Hedda 👍 🎉 

../Frenck
